packages/navigator/src/browser/navigator-contribution.ts

```diff
@@ -30,8 +30,14 @@ export class FileNavigatorContribution implements FrontendApplicationContributio
         this.fileNavigator.model.onSelectionChanged(selection =>
             this.selectionService.selection = selection
         );
-        this.workspaceService.root.then(fileStat => {
-            this.fileNavigator.model.root = DirNode.createRoot(fileStat);
+        this.workspaceService.rootResolved.then(resolved => {
```

Comment:
I really think adding `tryRoot` (`root` which can reject) instead of `rootResolved` is better API-wise, e.g. this code will be:
```ts
this.workspaceService.tryRoot().then(fileStat => 
    this.fileNavigator.model.root = DirNode.createRoot(fileStat)
).catch(() => 
    this.fileNavigator.update()
);
```
simpler contract, less to write and read

Reply:
All right, I am opened for a small modification; how should `WorkspaceRootAwareCommandHandler` handle when there is not root? Same applies to the terminal contributions too, for instance. The navigator example is clear.

Reply:
They continue using `this.root` as is.

I only meant that instead of checking `this.rootResolved` each time before `this.root`, it could be done once in `this.tryRoot` that returns `undefined` (or reject) if the root is not resolved. Returning `undefined` seems to be better looking at https://github.com/theia-ide/theia/pull/615/files#diff-416c9dea7bd5aec1cc7743b90b2c5528R51

Reply:
I see I got it. Let me try one more iteration on this.

Reply:
it would be:
```ts
        this.workspaceService.tryRoot().then(async resolvedRoot => {
            const root = resolvedRoot || (await this.fileSystem.getRoots())[0];
            if (root) {
...
            }
        });
```

Reply:
Yeah, it makes sense, I just did not check your snippet thoroughly.